### PR TITLE
[UI] Implemented category and description attributes for codes

### PR DESCRIPTION
**Pull request by hyperbx**
This PR extends the Codesv2 framework by adding two new keywords to the language for category and description attributes.

These keywords include;
- `in` for categories.
- `does` for descriptions.

These keywords can be used in any order, but `does` is recommended to be placed at the end.

Example usage;
```
Patch "Name" in "Category" by "Author" does "Description"
```

The description attribute also allows multiline descriptions;
```
Patch "Name" in "Category" by "Author" does
/*
Description
*/
```

The syntax used for multiline descriptions is to maintain compatibility with older versions of Hedge Mod Manager and newer code files, as old versions will read the multiline description as a multiline comment when compiling it as C#.

This PR includes categories and descriptions for all pre-existing codes, these have been tested on an older version of Hedge Mod Manager and do not cause any issues in terms of backwards compatibility.

UI changes;
- A new column has been added to the codes list for Category.
image
- Clicking a column in the codes list will sort the codes in either ascending or descending order based on the contents of it.
image
- Codes no longer have drag and drop support in the codes list.
- Columns can no longer be rearranged in the codes list.
- Double-clicking a code will open an information window containing the name, author and description attributes.
image
- Hovering over a code will display a tooltip to hint about double-clicking.
image

Future goals;
- Localisation support for code names, categories and descriptions (Codesv3).

### CODE REVIEW

File: HedgeModManager/Resources/Codesv2/SonicGenerations.hmm [hunk 1]
```diff
@@ -1,27 +1,29 @@
 !!VERSION 1.0
 
-Code "Infinite Rings" by "SuperSonic16"
+Code "Infinite Rings" in "Cheats" by "SuperSonic16"
 Write<uint>(GetPointer(0x1E5E2F0, 0x5B8), 0x3E7)
 
-Patch "Disable Ring Cap" by "Darío"
+Patch "Disable Ring Cap" in "Cheats" by "Darío" does "Allows the ring counter to exceed 999 rings."
```
Comment (RagdollClash) [inline HedgeModManager/Resources/Codesv2/SonicGenerations.hmm]: Wouldn't it be better for this to be under Gameplay?

**Comment by hyperbx** [comment]
Currently reworking how the codes list is sorted, will update when it's done.

**Comment by hyperbx** [comment]
Columns in the codes list will now sort by Category by default, with a new `CodesSortingColumnIndex` key added to the registry config so it remembers what your last sort was.

Codes now sort with the enabled codes first, then disabled codes last.

image

**Comment by HyperBE32** [comment]
Descriptions for codes are now displayed at the bottom of the codes list upon hovering over any that have a description. This can be disabled in the Settings tab.

aK1f5cZG1G

**Review by Sajidur78 — APPROVED** [review]
LGTM

**Pull request merged**